### PR TITLE
Add StorageClassBuilder to storage pkg

### PR DESCRIPTION
**Pull request by trewest**
Adds the StorageClassBuilder struct to the storage pkg

### CODE REVIEW

File: pkg/storage/storageclass.go [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+	// object is created.
+	errorMsg  string
+	apiClient *clients.Settings
+}
+
+// AdditionalOptions additional options for storageclass object.
+type AdditionalOptions func(builder *StorageClassBuilder) (*StorageClassBuilder, error)
+
+// NewStorageClassBuilder creates a new instance of StorageClassBuilder.
+func NewStorageClassBuilder(apiClient *clients.Settings, name, provisioner string) *StorageClassBuilder {
```
Comment (kononovn) [inline pkg/storage/storageclass.go:33]: I think it's name duplication. You can simply re-name it to NewClassBuilder because when you call this function you have storage in package name.
`storageClass := storage.NewClassBuilder`

File: pkg/storage/storageclass.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+	return builder, err
+}
+
+// Delete removes a storageclass.
+func (builder *StorageClassBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting storageclass %s",
```
Comment (kononovn) [inline pkg/storage/storageclass.go:202]: Could be single line.

File: pkg/storage/storageclass.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+		return err
+	}
+
+	builder.Object = nil
+
+	return err
+}
+
+// Update renovates the existing storageclass object with the storageclass definition in builder.
+func (builder *StorageClassBuilder) Update() (*StorageClassBuilder, error) {
```
Comment (kononovn) [inline pkg/storage/storageclass.go:222]: Have you managed to test this function? Does it work ?

**Review by kononovn — CHANGES_REQUESTED** [review]
Overall looks good, just few minor comments

**Review by cdvultur — APPROVED** [review]
/lgtm

File: pkg/storage/storageclass.go [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the storageclass is empty")
+
+		builder.errorMsg = "storageclass 'name' cannot be empty"
+	}
+
+	if provisioner == "" {
+		glog.V(100).Infof("The provisioner of the storageclass is empty")
```
Comment (achuzhoy) [inline pkg/storage/storageclass.go:55]: maybe set it yourself to "kubernetes.io/no-provisioner" instead.

Comment (trewest) [reply pkg/storage/storageclass.go:55]: Not sure I follow. kubernetes.io/no-provisioner is only one possible provisioner

File: pkg/storage/storageclass.go [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+}
+
+// WithParameter adds a parameter to the storage class definition.
+func (builder *ClassBuilder) WithParameter(parameterKey, parameterValue string) *ClassBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if parameterKey == "" {
+		glog.V(100).Infof("The parameterKey of the storageclass is empty")
```
Comment (achuzhoy) [inline pkg/storage/storageclass.go:114]: nit: there's no notion of parameterKey nor parameterValue for sc.
I'd rephrase to:
the "parameter's key" and "parameter's value" respectively

File: pkg/storage/storageclass.go (hunk 4 above)
Comment (achuzhoy) [reply pkg/storage/storageclass.go:55]: "no-provisioner" I'd consider using it as default maybe.

Comment (trewest) [reply pkg/storage/storageclass.go:55]: I personally think it should just be provided by the user. Otherwise the user needs some previous knowledge of what we have chosen the default to be which could be confusing.

**Review by achuzhoy — APPROVED** [review]
